scripts: Add MDS BLE gateway script (#8097)
nrfconnect/sdk-nrf: scripts/memfault/README.rst

Overview
********

The scripts scans for the :ref:`mds_readme` service UUID in advertising packets.
Once it founds the MDS service UUID, it connect to the remote device and reads MDS data.
Next the scripts waits for MDS notifications and forwards it to the URI read from the URI characteristics.

For more details about the Memfault check `Memfault SDK`_ and the :ref:`mod_memfault` to check details of the integration with |NCS|.

Requirements
************

The script source files are located in the :file:`scripts/memfault` directory.

To install the script's requirements, run the following command in its directory:

.. code-block:: console

   python3 -m pip install -r requirements.txt

This script also required the nRF52 series development kit to run the connectivity firmware that is programmed automatically when running the script.

Using the script
****************

Connect the nRF52 series development kit to your PC.
Run the script commands in the :file:`scripts/memfault` directory to start using the script.
Your DK will be programmed with connectivity firmware.
To list the script arguments, run the following command:

.. code-block:: console

   python3 mds_ble_gateway.py -h

> The following help information describes the available script arguments:

Reviewer: ```suggestion
The following help information describes the script arguments available:
```

.. code-block:: console

   Memfault BLE gateway

>    optional arguments:
>      -h, --help            show this help message and exit
>      --snr SNR             Segger chip id
>      --com COM             COM port name. For example COM0 or /dev/ttyACM0
>      --erase, -e           Erase target device before flashing firmware
>      --bond-disable        Disable bonding simulation
>      --reconnections RECONNECTIONS
>                            Number of reconnection attempts
>      --timeout TIMEOUT, -t TIMEOUT
>                            Connection establish timeout in seconds

Reviewer: ```suggestion
   optional arguments:
     -h, --help             Show this help message and exit
     --snr SNR             Segger chip ID
     --com COM             COM port name. For example COM0 or /dev/ttyACM0
     --erase, -e           Erase target device before flashing the firmware
     --bond-disable        Disable bonding simulation
     --reconnections RECONNECTIONS
                           Number of reconnection attempts
     --timeout TIMEOUT, -t TIMEOUT
                           Connection establish timeout in seconds
```
If these are already in the script help and you will not fix them in there, leave them as they are

Author: I can fix almost everything apart from `help` which is build-in

> To run script without additional configuration use a following command:

Reviewer: ```suggestion
To use the script without any additional configuration, run the following command:
```

> .. code-block:: console
> 
>    python3 mds_ble_gateway.py --snr <your DK Segger chip ID> --com <Serial port name>

Reviewer: ```suggestion
.. parsed-literal::

   python3 mds_ble_gateway.py --snr *your DK Segger chip ID* --com *Serial port name*
```

.. _mds_ble_gateway_script:

MDS BLE gateway script
######################

.. contents::
   :local:
   :depth: 2

The MDS BLE gateway is an application used as a client for the :ref:`mds_readme`.
It forwards data received on the Memfault Diagnostic Service to the Memfault`s cloud for further analysis.
The destination data URI and the Memfault project key are read from the service characteristics.

Overview
********

> The script scans for the :ref:`mds_readme` service UUID in the advertising packets.

Reviewer: ```suggestion
The script scans for the :ref:`mds_readme` UUID in the advertising packets.
```

Author: Done

Once it finds the MDS service UUID, it connects to the remote device and reads the MDS data.
The scripts waits for the MDS notifications and forwards them to the URI read from the URI characteristics.

For more details about the Memfault, see `Memfault SDK`_ and :ref:`mod_memfault` for details on integration with the |NCS|.

Requirements
************

The script source files are located in the :file:`scripts/memfault` directory.

To install the script's requirements, run the following command in its directory:

.. code-block:: console

   python3 -m pip install -r requirements.txt

This script also requires an the nRF52 Series development kit to run the connectivity firmware that is programmed automatically when running the script.

Using the script
****************

> 

Reviewer: Remove extra blank line.

Author: Done

1. Connect your nRF52 Series development kit to your PC.
#. Run the script commands in the :file:`scripts/memfault` directory.

Your DK will be programmed with the connectivity firmware.
To list the script arguments, run the following command:

.. code-block:: console

   python3 mds_ble_gateway.py -h

The following help information describes the script arguments available:

.. code-block:: console

   Memfault BLE gateway

   optional arguments:

>      -h, --help            show this help message and exit

Reviewer: ```suggestion
     -h, --help            Show this help message and exit
```

Author: It is printed by python argparser library so I don't want to change its default prompt
